ttnn/cpp/ttnn/operations/data_movement/transpose/device/transpose_program_factory.cpp

```diff
@@ -1899,7 +1915,11 @@ operation::ProgramWithCallbacks transpose_wh_multi_core_sharded(const Tensor& a,
 
         auto shard_spec = src_tensor.shard_spec().value();
 
-        uint32_t num_tiles_per_shard = shard_spec.numel() / TILE_HW;
+        const auto tile = src_tensor.get_tensor_spec().tile();
+        const uint32_t tile_hw = tile.get_tile_hw();
+        int32_t num_tiles = src_tensor.volume() / tile_hw;
+
+        uint32_t num_tiles_per_shard = shard_spec.numel() / tile_hw;
```

Comment:
definitely useful api for the spec so you don't have to write this again and again

Reply:
Yep